### PR TITLE
Scaling up involves creating a new-nodes inventory file.

### PR DESCRIPTION
**Pull request by mbruzek**
This is the approach I am taking for scaling up the OpenShift cluster. It is so far untested but please comment on the approach I am taking and let me know if that makes sense.

### CODE REVIEW

File: roles/openshift_on_openstack_scaleup/tasks/main.yml [hunk 1]
```diff
@@ -1,71 +1,122 @@
...
+    -i inventory
+    -i {{ inventory_py }}
+    {{ openshift_cluster_directory }}/provision.yml 2>&1 >> {{ openstack_scaleup_log }}
+  args:
+    # Use bash to get the posix style redirects.
+    executable: /bin/bash
+
+# Get the new OpenStack VMs by name.
+- name: Getting the new OpenStack VM network information by name
+  shell: "{{ openstack }} server list --name app-node-{{ item }}.scale-ci.example.com --format value -c Networks"
```
Comment (jmencak) [inline roles/openshift_on_openstack_scaleup/tasks/main.yml]: Can the domain be made into a variable?

File: roles/openshift_on_openstack_scaleup/tasks/main.yml [hunk 2]
```diff
@@ -1,71 +1,122 @@
...
-## Scaling up existing OpenShift deployment.
-#- name: Scale up the OpenShift resources
-#  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} openshift-ansible/playbooks/openshift-node/scaleup.yml 2>&1 > {{ openshift_scaleup_log }}"
-#  when: 'groups["new_nodes"] | list'
+# Scale up the OpenShift cluster.
+- name: Scaling up the OpenShift resources
+  shell: >
+    {{ ansible_playbook }} -vvv
+    --user openshift
+    -i inventory
```
Comment (jmencak) [inline roles/openshift_on_openstack_scaleup/tasks/main.yml]: I know this wasn't in the original code too, but the path to the inventory should also be ideally a var, similarly to inventory_py

File: roles/openshift_on_openstack_scaleup/tasks/main.yml (hunk 1 above)
Comment (mbruzek) [reply roles/openshift_on_openstack_scaleup/tasks/main.yml]: Yes good comment. We could construct the string from variables in all.yml but this code does not access all.yml at this time. I will have to think about this more.

**Comment by mbruzek** [comment]
I updated the code after some testing, still not 100% sure it will run because I ran into some openshift problems during scaleup.

File: roles/openshift_on_openstack_scaleup/tasks/main.yml (hunk 1 above)
Comment (mbruzek) [reply roles/openshift_on_openstack_scaleup/tasks/main.yml]: Created variables for these values in PR #37

**Comment by mbruzek** [comment]
I am merging this so we can test everything together in the next deploy.

File: roles/openshift_on_openstack_scaleup/tasks/main.yml [hunk 3]
```diff
@@ -1,71 +1,137 @@
...
+    -i {{ inventory_py }}
+    {{ openshift_cluster_directory }}/provision.yml 2>&1 >> {{ openstack_scaleup_log }}
+  args:
+    # Use bash to get the posix style redirects.
+    executable: /bin/bash
+
+# Get the new OpenStack VMs by name.
+- name: Getting the new OpenStack VM network information by name
```
Comment (jmencak) [inline roles/openshift_on_openstack_scaleup/tasks/main.yml]: This will be quite time-consuming.  At some point, we'll need to create the inventory faster.

Comment (mbruzek) [reply roles/openshift_on_openstack_scaleup/tasks/main.yml]: agreed. I had thought about getting listing of all the name and networks and parsing that list, but the Ansible task would have to loop over two different lists of different sizes in the same task (`with_items` from the output result and `with_sequence` to only add the new nodes).

**Pull request merged**